src/jloptions.c

```diff
@@ -108,8 +108,13 @@ static const char opts[]  =
     " -L, --load <file>         Load <file> immediately on all processors\n\n"
 
     // parallel options
-    " -t, --threads {N|auto}    Enable N threads; \"auto\" currently sets N to the number of local\n"
-    "                           CPU threads but this might change in the future\n"
+    " -t, --threads {N|auto}    Enable N threads; \"auto\" tries to infer a useful default number\n"
```

Comment:
I think you also need to edit `doc/src/manual/multi-threading.md` and `doc/src/manual/command-line-options.md` and `doc/man/julia.1`. And could also probably add a mention in NEWS.md